test/update_test.go

```diff
@@ -0,0 +1,145 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// TestUpdate is unit testing for the update operation.
+func TestUpdate(t *testing.T) {
+	ctx := context.Background()
+	client := newKine(t)
+
+	// Testing that update can create a new key if ModRevision is 0
+	t.Run("UpdateNewKey", func(t *testing.T) {
+		g := NewWithT(t)
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateNewKey"), "=", 0)).
+				Then(clientv3.OpPut("updateNewKey", "testValue")).
+				Else(clientv3.OpGet("updateNewKey", clientv3.WithRange(""))).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		{
+			resp, err := client.Get(ctx, "updateNewKey", clientv3.WithRange(""))
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Kvs).To(HaveLen(1))
+			g.Expect(resp.Kvs[0].Key).To(Equal([]byte("updateNewKey")))
+			g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue")))
+			g.Expect(resp.Kvs[0].ModRevision).To(BeNumerically("==", resp.Kvs[0].CreateRevision))
+		}
+	})
+
+	t.Run("UpdateExisting", func(t *testing.T) {
+		g := NewWithT(t)
+
+		var lastModRev int64
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateExistingKey"), "=", 0)).
+				Then(clientv3.OpPut("updateExistingKey", "testValue1")).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+			lastModRev = resp.Responses[0].GetResponsePut().Header.Revision
```

Comment:
We are accessing `resp.Responses[0]....` here, so I would add a test that `resp.Responses` is a list of length 1, along with basic validation of its values (if you consider any of them to be important here).